apps/aecore/lib/aecore/structures/spend_tx.ex

```diff
@@ -44,6 +45,9 @@ defmodule Aecore.Structures.SpendTx do
   defstruct [:receiver, :amount, :version]
   use ExConstructor
 
+  @spec get_chain_state_name() :: Account.chain_state_name()
+  def get_chain_state_name(), do: :accounts
```

Comment:
(important) This should be `nil` otherwise `process_chainstate` will receive `chainstate.accounts` twice and will supposed to be returning new accounts state twice. This passes tests currently due to the order of updating in `DataTx`.

Reply:
Good find, I proposed this to @d-velev but had some additional code to catch this. `:accounts` returned is the semantic correct behavior but it should be filtered.

https://github.com/aeternity/elixir-node/blob/GH-274/apps/aecore/lib/aecore/structures/data_tx.ex#L83-L88